src/middleware/express.ts

```diff
@@ -0,0 +1,123 @@
+/*!
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as bunyan from 'bunyan';
+import {EWOULDBLOCK} from 'constants';
+import {NextFunction, Request, Response} from 'express';
+
+import {LOGGING_TRACE_KEY, LoggingBunyan} from '../index';
+import * as types from '../types/core';
+
+import * as context from './context';
+
+export interface LoggingMiddlewareFunction {
+  (req: Request, res: Response, next: NextFunction): void;
+  logger: ReturnType<typeof bunyan.createLogger>;
+  mw: LoggingMiddlewareFunction;
+}
+
+export type LogObject = {
+  // tslint:disable-next-line:no-any bunyan interface.
+  [level in bunyan.LogLevelString]: (...args: any[]) => void;
+};
+
+type AnnotatedRequest = Request&{
+  log: LogObject;
+};
+
+function makeLogFunction(
+    level: bunyan.LogLevelString,
+    logger: ReturnType<typeof bunyan.createLogger>, trace: string|null) {
+  // tslint:disable-next-line:no-any bunyan interface.
+  return (...args: any[]) => {
+    let fields;
```

Comment:
no type annotation

Reply:
its gotta be `any`.  No point in annotating that.